src/mem/metaslab.h

```diff
@@ -89,20 +74,51 @@ namespace snmalloc
      */
     uintptr_t remote_and_sizeclass{0};
 
+    /**
+     * Constructor from two pointer-sized words.  The subclass is responsible
+     * for ensuring that accesses to these are type-safe.
+     */
+    constexpr MetaEntryBase(uintptr_t m, uintptr_t ras)
+    : meta(m), remote_and_sizeclass(ras)
+    {}
+
+    /**
+     * Default constructor, zero initialises.
+     */
+    constexpr MetaEntryBase() : MetaEntryBase(0, 0) {}
+
   public:
-    constexpr MetaEntry() = default;
+    /**
+     * When a meta entry is in use by the back end, it exposes tow words of
+     * state.  The low bits in both are reserved.  Bits in this bitmask must
+     * not be set by the back end in either word.
+     *
+     * During a major release, this constraint may be weakened, allowing the
+     * back end to set more bits.
+     */
+    static constexpr address_t BackendReservedMask =
```

Comment:
We should probably have a style guide if we want to have rules like that because it's not a convention I've seen anywhere else.  The more common convention is that *macros* are all-caps, and I find it confusing in snmalloc that we have macros and non-macro constants all uppercase.  I've made the change but it makes me sad.

Reply:
A style guide is a good idea.  Let's do it in a different PR ;-)